index.js

```diff
@@ -187,6 +187,11 @@ exports.evm2wast = function (evmCode, opts = {
       segmentStackLow = segmentStackDeta
     }
 
+    // creates a stack trace
+    if (opts.stackTrace) {
+      segment += `(call $stackTrace (i32.const ${pc}) (i32.const ${opint}) (i32.const ${gasCount}) (get_global $sp))\n`
```

Comment:
I think the stack pointer will be broken if it is done before? Needs to be fixed however.

Reply:
I'm pretty sure it's correct. I looked into this earlier on the call with Casey, and also mentioned it in the commitlog. Here's the trace output as is, which looks correct:

```
Executing...
importGlobals
useGas 12
evmTrace
{"depth":0,"gas":378988,"gasCost":3,"op":96,"pc":0,"stack":[]}
evmTrace
{"depth":0,"gas":378988,"gasCost":6,"op":96,"pc":2,"stack":["0x0000000000000000000000000000000000000000000000000000000000000001"]}
evmTrace
{"depth":0,"gas":378988,"gasCost":9,"op":1,"pc":4,"stack":["0x0000000000000000000000000000000000000000000000000000000000000001","0x0000000000000000000000000000000000000000000000000000000000000001"]}
evmTrace
{"depth":0,"gas":378988,"gasCost":12,"op":96,"pc":5,"stack":["0x0000000000000000000000000000000000000000000000000000000000000002"]}
evmTrace
{"depth":0,"gas":378988,"gasCost":12,"op":85,"pc":7,"stack":["0x0000000000000000000000000000000000000000000000000000000000000000","0x0000000000000000000000000000000000000000000000000000000000000002"]}
storageStore 20 0
```

If I initialize `$sp` at zero rather than at -32, it adds an extra 0x00 onto the front of the stack in all cases, which seems wrong.

Reply:
Oh right, the client just ignores anything `<0`. Perhaps worth documenting that in the `evmTrace` "documentation".

Reply:
> Oh right, the client just ignores anything <0

I think I'm making the case that, by the time the first `stackTrace` runs, it _is_ zero.

> Perhaps worth documenting that in the evmTrace "documentation".

Do you mean a comment in this code?

Reply:
I think it is the README of Hera which has it at the moment.